austation/code/game/materialization.dm

```diff
@@ -0,0 +1,291 @@
+/obj/structure/table/mat_shiritori
+	name = "\improper Materialization Shiritori"
+	desc = "A magical Shiritori table capable of temporarily materializing any object that exists in this reality."
+	icon = 'austation/icons/obj/structures/magic_table.dmi'
+	icon_state = "magic_table"
+	max_integrity = 300
+	canSmoothWith = null
+	flags_1 = NODECONSTRUCT_1
+	var/current_letter = ""
+	var/time_left = 40 SECONDS
+	var/active = FALSE
+	var/game_starting = FALSE // prep period
+	var/start_time = 15 // how many seconds people get to opt in
+	var/turns = 0 // how many turns have passed
+	var/static/list/atom_list // starts off as empty, we only want to make this if we have a table, otherwise we're just wasting memory
+	var/list/countdown_warns = list(50, 100, 150) // used in process(), list contains at what times we should announce time_left.
+	var/list/spent_objs = list() // typepaths that have been used before
+	var/list/entities = list() // spawned entities that are currently in existance
+	var/list/players = list()
+	var/mob/living/carbon/human/current_player
+	var/list/knockouts = list()
+	var/static/list/blacklist // handled in setup_blacklist(). Contains root paths that aren't supposed to be spawned (Yes, a bit of a pain to maintain, but it's not the end of the world if someone forgets to update it. The alternative is having a variable on every damn datum)
+
+/obj/structure/table/mat_shiritori/Destroy()
+	end_game()
+	return ..()
+
+/obj/structure/table/mat_shiritori/proc/prepare_game()
+	if(active || game_starting)
+		return
+	setup_blacklist()
+	players.len = 0
+	knockouts.len = 0
+	time_left = initial(time_left)
+	say("Game starting in [start_time] seconds. Place your hand on the table to play.")
+	start_game()
+
+//  Called after everything is ready
+/obj/structure/table/mat_shiritori/proc/start_game()
+	set waitfor = FALSE
+	game_starting = TRUE
+	for(var/i in 1 to start_time)
+		var/countdown = start_time - (i - 1)
```

Comment:
you could probably just use a standard countdown for loop instead of counting up, and then doing subtraction, but I won't force that to be changed since it's very minor

Reply:
I originally had it as that but it broke something and I can't remember what.